File: updated_code/code/python/1763-longest-nice-substring.py
Python
class LongerNiceSubstring:
    """
    This class contains a method to find the longest nice substring from a given string.

    A string `s` is **nice** if, for every letter of the alphabet that `s` contains,
    it appears **both** in uppercase and lowercase.

    Args:
    self: The instance of the class.
    s (str): The input string.

    Returns:
    str: The longest **substring** of `s` that is **nice**.
    """
    def longest_nice_substring(self, s: str) -> str:
        if not s:
            return ""

        max_len = 0
        res = ""

        for i in range(len(s)):
            tmp_set = set()
            for j in range(i, len(s)):
                c = s[j]
                if c.lower() in tmp_set:
                    break
                if c not in tmp_set and c.lower() not in tmp_set:
                    tmp_set.add(c.lower())
                    tmp_set.add(c)
                if len(tmp_set) == 26:
                    if j - i + 1 > max_len:
                        max_len = j - i + 1
                        res = s[i:j+1]

        return res

File: updated_code/code/java/1763-longest-nice-substring.java
package com.solution;

/**
 * Class comments are required.
 */
public class Solution {

    /**
     * Public method and constructor comments are required.
     * 
     * @param s Given string
     * @return The longest **substring** of `s` that is **nice**.
     */
    public String longestNiceSubstring(String s) {
        final int n = s.length();
        char[] ca = s.toCharArray();
        int res = 0, resStart = 0;
        for (int i = 0; i < n; i++) {
            boolean isNice = true;
            for (char c = 'a'; c <= 'z'; c++) {
                final int pos = ca[n - 1 - i + n - (int) c] - c;
                if (pos < 0) continue;
                if ((ca[i + pos] &= ca[i + pos] - c) % 32 != 0) {
                    isNice = false;
                    break;
                }
            }
            if (!isNice) {
                while (i < n) {
                    if ((ca[i] & 1) != (ca[i] / 2 | ca[i] % 32)) {
                        break;
                    }
                    i++;
                }
            } else {
                res = Math.max(res, i - resStart + 1);
                if (res >= resStart + i) {
                    resStart = i - res;
                }
            }
        }
        return s.substring(resStart, resStart + res);
    }
}